editors/code/src/source_change.ts

```diff
@@ -55,3 +55,13 @@ export async function applySourceChange(ctx: Ctx, change: SourceChange) {
         );
     }
 }
+
+export async function applySourceChange(ctx: Ctx, change: SourceChange, alternativeChanges: SourceChange[] | undefined) {
```

Comment:
It still seems to me that overloading this function is wrong. Depending on the arguments you pass to it, it can be either just apply some side-effect, or it can interactively ask user for some input. I'd rather avoid mixing interactive and non-interactive usages in the same function.

Reply:
Yeah, the client side split is great, I like it.
The question is how to pass the data from a backend, that what I wanted to discuss in 
https://github.com/rust-analyzer/rust-analyzer/pull/2716#discussion_r362313985

So far it seems that I'd better go with the option #2 from https://github.com/rust-analyzer/rust-analyzer/pull/2716#discussion_r363812239

Reply:
I think you'll need to pick a different action name in conv_rs, but other than that, the current backend setup looks reasonable to me. 

I can't say that I feel that it is unquestionably perfect, but I can't suggest any immediate obvious improvements :)

Reply:
Ok, I've come up with the changes that hopefully make the overall thing better.